tonic/src/transport/service/timeout.rs

```diff
@@ -0,0 +1,251 @@
+use http::Request;
+use http::{HeaderMap, HeaderValue};
+use pin_project::pin_project;
+use std::{fmt, pin::Pin};
+use std::{
+    future::Future,
+    task::{Context, Poll},
+    time::Duration,
+};
+use tokio::time::Sleep;
+use tower_service::Service;
+
+#[derive(Debug, Clone)]
+pub(crate) struct Timeout<S> {
```

Comment:
Might make sense to call this `GrpcTimeout` since it parses a header for that :D

```suggestion
pub(crate) struct GrpcTimeout<S> {
```